taca/nanopore/promethion_transfer.py

```diff
@@ -109,13 +113,152 @@ def archive_finished_run(run_dir, archive_dir):
     else:
         print("Some data is still left in {}. Keeping it.".format(top_dir))  # Might be another run for the same project
 
+
+def parse_position_logs(minknow_log_dir: str) -> list:
+    """Look through all flow cell position logs and boil down into a strucutred list of dicts
+
+    Example output:
+    [{
+        "timestamp": "2023-07-10 15:44:31.481512",
+        "category": "INFO: platform_qc.report (user_messages)",
+        "body": {
+            "flow_cell_id": "PAO33763"
+            "num_pores": "8378"
+        }
+    } ... ]
+
+    """
+
+    position_log_file_name = "control_server_log-0.txt"
+    log_timestamp_format = "%Y-%m-%d %H:%M:%S.%f"
+
+    positions = []
+    for col in "123":
+        for row in "ABCDEFGH":
+            positions.append(col + row)
+
+    entries = []
+    for position in positions:
+        with open(
+            f"{minknow_log_dir}/{position}/{position_log_file_name}", "r"
+        ) as stream:
+            lines = stream.readlines()
+            for i in range(0, len(lines)):
+                line = lines[i]
+                if line[0:4] != "    ":
+                    # Line is log header
+                    split_header = line.split(" ")
+                    timestamp = " ".join(split_header[0:2])
+                    category = " ".join(split_header[2:])
+
+                    entry = {
+                        "position": position,
+                        "timestamp": timestamp.strip(),
+                        "category": category.strip(),
+                    }
+                    entries.append(entry)
+                else:
+                    # Line is log body
+                    if "body" not in entry:
+                        entry["body"] = {}
+                    key = line.split(": ")[0].strip()
+                    val = ": ".join(line.split(": ")[1:]).strip()
+                    entry["body"][key] = val
+
+    return entries
+
+
+def get_pore_counts(position_logs: list) -> list:
+    """Take the flowcell log list output by parse_position_logs() and subset to contain only QC and MUX info."""
+
+    pore_counts = []
+    for entry in position_logs:
+
+        if "INFO: platform_qc.report (user_messages)" in entry["category"]:
+            type = "qc"
+        elif "INFO: mux_scan_result (user_messages)" in entry["category"]:
+            type = "mux"
+        else:
+            type = "other"
+
+        if type in ["qc", "mux"]:
+
+            new_entry = {
+                "flow_cell_id": entry["body"]["flow_cell_id"],
```

Comment:
Will the entries always have a body if they are qc or mux? There was a `if "body" not in entry:` in `parse_position_logs`

Reply:
Yes, some logs are one-liners and do not have a body but QC and MUX always seem to.